staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -48,3 +48,63 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown `json:"configuration"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// connectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService `json:"connectionServices"`
```

Comment:
This makes sense for a file on disk.  When/if this goes onto the cluster, we probably want to actually have multiple named objects.